plugins/trace_api_plugin/include/eosio/trace_api/chain_extraction.hpp

```diff
@@ -77,13 +70,19 @@ class chain_extraction_impl_type {
       store_lib( block_state );
    }
 
+   void on_block_start( uint32_t block_num ) {
+      cached_traces.clear();
```

Comment:
I like the pattern used in `state_history_plugin` where we have a method that clears all caches called from the `block_start` handler and after processing a `block_state_ptr` on lines 99-100

Reply:
Done.